### DISCUSSION TITLE
scrambles by http request

### DISCUSSION THREAD

**Discussion by njarizas** [thewca/tnoodle#636] [Q&A]
Hello, I would like to know if in the TNoodle-WCA-1.0.1.jar version there is a way to obtain the scrambles through http request since I have a web utility to practice and take my times and I obtain the mixes from the path http://localhost:2014/scramble/.txt?=333 but in the latest version it no longer works for me

**Comment by gregorbg** [comment]
Hey, indeed the API has changed in the TNoodle 1.0 release. It is not possible right now to get scrambles directly, because the TNoodle JAR that you can download adheres to the WCIF specification.

If you post a valid WCIF to `http://localhost:2014/wcif/scrambles`, TNoodle will fill in the scrambles and return the WCIF including scrambles for you. If you need access to scrambles outside the context of WCIF, feel free to post an issue and describe your use-case so we can consider adding a cleaner API in the future!

**Comment by campos20** [comment]
Maybe this will help. It's a personal project, deployed to heroku (so, first load can be slow because heroky put containers on sleep mode).

https://scramble-web-api.herokuapp.com/swagger-ui.html

**Reply by njarizas** [reply]
Hello, the proposed solution could be useful to me, however I prefer to continue using TNoodle-WCA-0.15.1.jar for my purposes.
Thanks for your help
